File: ThreeDSecure/src/main/java/com/braintreepayments/api/threedsecure/ThreeDSecurePaymentAuthResult.java
package com.braintreepayments.api.threedsecure;

import com.cardinalcommerce.cardinalmobilesdk.models.ValidateResponse;

public class ThreeDSecurePaymentAuthResult {

    private final String jwt;
    private final ValidateResponse validateResponse;
    private final ThreeDSecureParams threeDSecureParams;

    private final Exception error;

    ThreeDSecurePaymentAuthResult(ThreeDSecureParams threeDSecureParams, String jwt,
                                  ValidateResponse validateResponse) {
        this.jwt = jwt;
        this.validateResponse = validateResponse;
        this.threeDSecureParams = threeDSecureParams;
        this.error = null;
    }

    ThreeDSecurePaymentAuthResult(Exception error) {
        this.error = error;
        this.jwt = null;
        this.validateResponse = null;
        this.threeDSecureParams = null;
    }

    Exception getError() {
        return error;
    }

    ThreeDSecureParams getThreeSecureResult() {
        return threeDSecureParams;
    }

    ValidateResponse getValidateResponse() {
        return validateResponse;
    }

    String getJWT() {
        return jwt;
    }
}

File: ThreeDSecure/src/main/java/com/braintreepayments/api/threedsecure/ThreeDSecurePaymentAuthResult.kt
package com.braintreepayments.api.threedsecure

//import com.cardinalcommerce.cardinalmobilesdk.models.ValidateResponse

/**
 * Result returned from [ThreeDSecureLauncher.launch]
 *
 * @property jwt jwt value
// * @property validateResponse validate response data from Cardinal
 * @property threeDSecureParams additional parameters from the auth request
 * @property error error if the auth request failed
 */
data class ThreeDSecurePaymentAuthResult internal constructor(
    internal val jwt: String? = null,
//    internal val validateResponse: ValidateResponse? = null,
    internal val threeDSecureParams: ThreeDSecureParams? = null,
    internal val error: Exception? = null
)
